src/platform/packages/shared/kbn-alerts-ui-shared/src/alert_filter_controls/filter_group.tsx

```diff
@@ -234,14 +230,26 @@ export const FilterGroup = (props: PropsWithChildren<FilterGroupProps>) => {
       next: debouncedFilterUpdates,
     });
 
+    return () => {
+      filterChangedSubscription.current?.unsubscribe();
```

Comment:
Since filterChangedSubscription ref is never used outside the scope of the use effect, there is no need for a ref. How about just

```
useEffect(() => {
    if (!controlGroup) {
      return;
    }

    const subscription = controlGroup.filters$.subscribe({
      next: debouncedFilterUpdates,
    });

    return () => {
      subscription.unsubscribe();
    };
  }, [controlGroup, debouncedFilterUpdates]);
```